### ISSUE TITLE
Testing Issues

### ISSUE THREAD

**Issue by 786039663** [tencent-ailab/IP-Adapter#164]
After I converted the model.safetensors file I got after training to ip_adapter.bin, when I tested it, it reported an error.
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/33db8f22-3c7f-407c-abd1-52afcde5b9d0)
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/7bc24e13-d531-41dd-ace3-5aba578795aa)

**Comment by xiaohu2015** [comment]
do you use the code in https://github.com/tencent-ailab/IP-Adapter#how-to-train?

**Comment by 786039663** [comment]
Yes, I trained according to this https://github.com/tencent-ailab/IP-Adapter#how-to-train.
After training, get model.safetensors. Then convert model.safetensors to ip_adapter.bin
Finally an error when loading ip_adapter.bin
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/e3e453fa-0472-4980-bbda-249dda75bad9)
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/2a6d161b-8c45-4a6d-b9ba-1eaa622bf44b)
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/26e3af68-070a-46a4-8da8-9a7c69d967e5)
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/6400822a-04e6-4833-ae89-bf4206c750a8)
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/233ce8f0-283c-4b26-aec2-1a181d29484b)

**Comment by xiaohu2015** [comment]
you can load the `ip_adapter.bin`, and print the keys:
```
import torch

sd = torch.load("ip_adapter.bin")

print(sd["image_proj"].keys())
print(sd["ip_adapter"].keys())
```

**Comment by 786039663** [comment]
That's the result, but I don't know what it means.
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/87febde6-eabb-4d8c-b1b4-022ec472f2b5)

**Comment by 786039663** [comment]
The training was prompted with this message.
![image](https://github.com/tencent-ailab/IP-Adapter/assets/152965055/cb74d7b2-7055-4f72-bf90-ad2aa6275cd5)

**Comment by xiaohu2015** [comment]
you can try this:
```python
import torch
ckpt = "checkpoint-50000/pytorch_model.bin"

sd = torch.load(ckpt, map_location="cpu")
image_proj_sd = {}
ip_sd = {}
for k in sd:
    if k.startswith("unet"):
        pass
    elif k.startswith("image_proj_model"):
        image_proj_sd[k.replace("image_proj_model.", "")] = sd[k]
    elif "_ip." in k:
        ip_sd[k] = sd[k]
```
torch.save({"image_proj": image_proj_sd, "ip_adapter": ip_sd}, "ip_adapter.bin")

**Comment by masaisai111** [comment]
Have you solved the problem

> 系统提示培训并显示此消息。 ![图像](https://private-user-images.githubusercontent.com/152965055/288024800-cb74d7b2-7055-4f72-bf90-ad2aa6275cd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3MzE3NzksIm5iZiI6MTcxMjczMTQ3OSwicGF0aCI6Ii8xNTI5NjUwNTUvMjg4MDI0ODAwLWNiNzRkN2IyLTcwNTUtNGY3Mi1iZjkwLWFkMmFhNjI3NWNkNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMFQwNjQ0MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzNkNDFkNWYwOWI4M2JlNDdjOGRlYzQ2NzdkYzBjMDE3N2YyZjM3YzE3ZTFhM2MxNWNhMGNmYTE1N2FkZGFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UXBLe1Qg1mDA9Z5rCWxzxUF2So0OofnE_rcLM9yOvS0)

**Comment by unl1002** [comment]
You should change this line of code.
`accelerator.save_state(save_path, safe_serialization=False)`
